internal/pipe/build/build.go

```diff
@@ -113,6 +118,79 @@ func runPipeOnBuild(ctx *context.Context, build config.Build) error {
 	return g.Wait()
 }
 
+func proxy(ctx *context.Context, build config.Build) (config.Build, error) {
+	if !build.IsProxied() {
+		return build, nil
+	}
+
+	template := tmpl.New(ctx)
+
+	proxy, err := template.Apply(build.Proxy.Path)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	version, err := template.Apply(build.Proxy.Version)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	log := log.WithField("id", build.ID)
+	log.Infof("proxying %s@%s", proxy, version)
+
+	template = template.WithExtraFields(tmpl.Fields{
+		"Proxy":   proxy,
+		"Version": version,
+	})
+
+	mod, err := template.Apply(`
+module {{ .ProjectName }}
+
+require {{ .Proxy }} {{ .Version }}
+
+`)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	main, err := template.Apply(`
+// +build main
+package main
+
+import _ "{{ .Proxy }}"
+`)
+	if err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	dir := fmt.Sprintf("%s/build_%s", ctx.Config.Dist, build.ID)
+
+	log.Debugf("creating needed files")
+
+	if err := os.Mkdir(dir, 0o755); err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	if err := os.WriteFile(dir+"/main.go", []byte(main), 0o650); err != nil {
+		return build, fmt.Errorf("failed to proxy module: %w", err)
+	}
+
+	if err := os.WriteFile(dir+"/go.mod", []byte(mod), 0o650); err != nil {
```

Comment:
The real gold standard here would be to also copy the `go.sum` file from the module in question and use that (before any builds). That will establish consistency with the project in terms of expectations on checksums.

Reply:
I've updated https://github.com/goreleaser/goreleaser/issues/1354#issuecomment-791175542 with that step.

Reply:
makes sense